### PR TITLE
Update version for the next release (v0.8.0)

### PR DESCRIPTION
**Pull request by alallema**
This version includes a major redesign of the SDK. ([#425](https://github.com/meilisearch/meilisearch-java/pull/425)) [@alallema](https://github.com/alallema)

## ⚠️ Breaking changes

* Redesign of the client (#449) @alallema:
    * Use `OkHttp` library by default for the Meilisearch client
    * No interface for the creation of a Client disappearing of the class `GenericServiceTemplate`, `ServiceTemplate`, `AbstractClient`, `ApacheClient`, and `DefaultHttpClient`.
* Rename `getAllIndexes` method in `getIndexes` (#477) @alallema
* All parameters of the managers accessible in the client are now private: `Config`, `IndexesHandler`, `InstanceHandler`, ` TasksHandler`, `KeysHandler`, `JsonHandler`.
* Factories for the answer and the response disappeared, those classes were removed: `BasicRequestFactory`, `BasicHttpResponse`, `BasicHttpRequest`, and `MeilisearchHttpRequest`.
* Rewriting of the JsonHandler
    * Offering the possibility to use `Gson`, `Jackson` or create your own handler.([#432]
    * Remove `JsonbJsonHandler`.
* Dump returns a `Task`from v0.28.0 so the `createDump` method has been removed just as the `DumpHandler` class.
* Renaming class `Details` in `TaskDetails`
* All methods return now a `MeilisearchException` instead of a `Exception`.

## 🚀 Enhancements

* Replaced traditional getter setter by `@Getter` `@Setter` from Lombok library ([#385](https://github.com/meilisearch/meilisearch-java/pull/385)) [@ghousek1](https://github.com/ghousek1)
* Improve Docker configuration in the package ([#399](https://github.com/meilisearch/meilisearch-java/pull/399))
* Add code-coverage tool (jacoco) ([#422](https://github.com/meilisearch/meilisearch-java/pull/422)) [@brunoocasali](https://github.com/brunoocasali)
* Refactoring:
    * Rewriting of the Error Handler ([#438](https://github.com/meilisearch/meilisearch-java/pull/438)) [@alallema](https://github.com/alallema)
        * Like the other SDKs, this one now contains `MeilisearchApiError`, `MeiliSearchCommunicationError`, `MeilisearchTimeoutError`, `JsonDecodingException` as well as `JsonEncodingException`.
        * All methods return now a `MeilisearchException` instead of a Exception.
(https://github.com/meilisearch/meilisearch-java/pull/432))  [@alallema](https://github.com/alallema)
    * Rewrite some missing method (#473) [@alallema](https://github.com/alallema)
        * health()
        * isHealthy()
        * getVersion()
        * getStats()
        * index.getStats()
        * updateKey() ([#476](https://github.com/meilisearch/meilisearch-java/pull/476)) [@alallema](https://github.com/alallema)
    * Add typo tolerance settings ([#371](https://github.com/meilisearch/meilisearch-java/pull/371)) [@alallema](https://github.com/alallema)
    * Add toString method to SearchRequest Class ([#451](https://github.com/meilisearch/meilisearch-java/pull/451)) [@alallema](https://github.com/alallema)
    * Add support to PATCH HTTP method ([#472](https://github.com/meilisearch/meilisearch-java/pull/472)) [@alallema](https://github.com/alallema)

Thanks again to @alallema,  @brunoocasali,   @ghousek1, @kisaga ! 🎉

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -87,6 +87,7 @@ test {
 	}
 	testLogging {
 		events 'passed', 'skipped', 'failed'
+		showStandardStreams = true
```
Comment (brunoocasali) [inline build.gradle:90]: This should be committed?

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -401,15 +401,15 @@ getting_started_add_documents_md: |-
   <dependency>
     <groupId>com.meilisearch.sdk</groupId>
     <artifactId>meilisearch-java</artifactId>
-    <version>0.7.2</version>
```
Comment (brunoocasali) [inline .code-samples.meilisearch.yaml:404]: I also think you can add to the changelog that you removed some classes like `GenericServiceTemplate`, `Dump`, `DumpHandler`, `MeilisearchHttpRequest`, and others...  

The users will search your CHANGELOG for keywords like "why the class I was using does not exist anymore?". "Now this is the new way to use the methodABC" and so on (like you did for the exception classes)...

**Review by brunoocasali — CHANGES_REQUESTED** [review]

**Review by brunoocasali — APPROVED** [review]
@alallema let's merge this! 🎉

**Comment by alallema** [comment]
bors merge

**Pull request merged**